arc_solvers/models/qa/multi_choice/qa_multi_choice_max_att.py

```diff
@@ -0,0 +1,231 @@
+
+    question_encoded = context_enc(question_words)  # context encoder can be any AllenNLP supported or None
+    choice_encoded = context_enc(choice_words)
+
+    question_aggregate = aggregate_method(question_encoded)
+    choice_aggregate = aggregate_method(choice_encoded)
+
+    inter = concat([question_aggregate, choice_aggregate, abs(choice_aggregate - question_aggregate), question_aggregate * choice_aggregate)
+
+    The model is a popular baseline but it is most simialar to Conneau, A. et al. (2017) ‘Supervised Learning of Universal Sentence Representations from Natural Language Inference Data’,
+    without the feed-forward layer!
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``
+    text_field_embedder : ``TextFieldEmbedder``
+        Used to embed the ``question`` and ``choice`` ``TextFields`` we get as input to the
+        model.
+    aggregate_feedforward : ``FeedForward``
+        These feedforward networks are applied to the concatenated result of the
+        encoder networks, and its output is used as the entailment class logits.
+    question_encoder : ``Seq2SeqEncoder``, optional (default=``None``)
+        After embedding the question, we can optionally apply an encoder.  If this is ``None``, we
+        will do nothing.
+    choice_encoder : ``Seq2SeqEncoder``, optional (default=``None``)
+        After embedding the choice, we can optionally apply an encoder.  If this is ``None``,
+        we will use the ``question_encoder`` for the encoding (doing nothing if ``question_encoder``
+        is also ``None``).
+    initializer : ``InitializerApplicator``, optional (default=``InitializerApplicator()``)
+        Used to initialize the model parameters.
+    regularizer : ``RegularizerApplicator``, optional (default=``None``)
+        If provided, will be used to calculate the regularization penalty during training.
+    share_encoders : ``bool``, optional (default=``false``)
+        Shares the weights of the question and choice encoders.
+    aggregate_question : ``str``, optional (default=``max``, allowed options [max, avg, sum, last])
+        The aggregation method for the encoded question.
+    aggregate_choice : ``str``, optional (default=``max``, allowed options [max, avg, sum, last])
+        The aggregation method for the encoded choice.
+    """
+
+    def __init__(self, vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,
+                 question_encoder: Optional[Seq2SeqEncoder] = None,
+                 choice_encoder: Optional[Seq2SeqEncoder] = None,
+                 initializer: InitializerApplicator = InitializerApplicator(),
+                 aggregate_question: Optional[str] = "max",
+                 aggregate_choice: Optional[str] = "max",
+                 embeddings_dropout_value: Optional[float] = 0.0,
+                 params=Params) -> None:
```

Comment:
This `params` argument has the `Params` _type_ as a default, which I think should be `Params({})`. It is also not optional, as you require that it contains some information about `att_question_to_choice `.